src/test/java/com/jcabi/github/StatusesTest.java

```diff
@@ -0,0 +1,117 @@
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.github;
+
+import com.google.common.base.Optional;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link Statuses}.
+ * @author Chris Rebert (github@chrisrebert.com)
+ * @version $Id$
+ * @since 0.24
+ */
+public final class StatusesTest {
+    /**
+     * Test status URL.
+     */
+    private static final String URL = "http://status.jcabi-github.invalid/42";
+
+    /**
+     * Test commit status context string.
+     */
+    private static final String CONTEXT = "jcabi/github/test";
+
+    /**
+     * StatusCreate can convert itself to JSON.
+     */
+    @Test
+    public void convertsToJsonWhenAllPresent() {
+        final String success = "Everything is awesome";
+        MatcherAssert.assertThat(
+            new Statuses.StatusCreate(Status.State.SUCCESS)
+                .withTargetUrl(Optional.of(URL))
+                .withDescription(success)
+                .withContext(Optional.of(CONTEXT))
+                .json().toString(),
+            // @checkstyle LineLength (1 line)
+            Matchers.equalTo("{\"state\":\"success\",\"description\":\"Everything is awesome\",\"context\":\"jcabi/github/test\",\"target_url\":\"http://status.jcabi-github.invalid/42\"}")
+        );
+    }
+
+    /**
+     * StatusCreate can convert itself to JSON when it has no URL.
+     */
+    @Test
+    public void convertsToJsonWhenUrlAbsent() {
+        final String success = "Living the dream!";
+        MatcherAssert.assertThat(
+            new Statuses.StatusCreate(Status.State.SUCCESS)
+                .withDescription(success)
+                .withContext(Optional.of(CONTEXT))
+                .json().toString(),
+            // @checkstyle LineLength (1 line)
+            Matchers.equalTo("{\"state\":\"success\",\"description\":\"Living the dream!\",\"context\":\"jcabi/github/test\"}")
+        );
+    }
+
+    /**
+     * StatusCreate can convert itself to JSON when it has no description.
+     */
+    @Test
+    public void convertsToJsonWhenDescriptionAbsent() {
+        MatcherAssert.assertThat(
+            new Statuses.StatusCreate(Status.State.FAILURE)
+                .withTargetUrl(Optional.of(URL))
+                .withContext(Optional.of(CONTEXT))
+                .json().toString(),
+            // @checkstyle LineLength (1 line)
+            Matchers.equalTo("{\"state\":\"failure\",\"description\":\"\",\"context\":\"jcabi/github/test\",\"target_url\":\"http://status.jcabi-github.invalid/42\"}")
+        );
+    }
+
+    /**
+     * StatusCreate can convert itself to JSON when it has no context.
+     */
+    @Test
+    public void convertsToJsonWhenContextAbsent() {
+        final String pending = "Kragle is drying...";
```

Comment:
@carlosmiranda This will no longer be single-use after the switch to `JsonObjectBuilder`.